cg/meta/upload/gisaid/gisaid.py

```diff
@@ -114,92 +142,117 @@ def get_gisaid_samples(self, case_id: str) -> list[GisaidSample]:
-            LOG.info("GISAID log exists in case bundle in Housekeeper")
+    def create_and_include_gisaid_log_file_to_hk(self, case_id: str) -> None:
+        """Create and include GISAID log file to a Housekeeper."""
+        if _ := self.housekeeper_api.get_files(
+            bundle=case_id, tags=[GisaidTag.LOG, case_id]
+        ).first():
+            LOG.info(f"GISAID log exists in case: {case_id} bundle in Housekeeper")
             return
 
-        log_file_path = Path(self.gisaid_log_dir, case_id).with_suffix(".log")
-        if not log_file_path.parent.exists():
-            raise ValueError(f"Gisaid log dir: {self.gisaid_log_dir} doesnt exist")
-        if not log_file_path.exists():
-            log_file_path.touch()
+        log_file = Path(self.gisaid_log_dir, case_id).with_suffix(FileExtensions.LOG)
+        if not log_file.parent.exists():
+            raise ValueError(f"GISAID log dir: {self.gisaid_log_dir} does not exist")
+        if not log_file.exists():
+            log_file.touch()
         self.housekeeper_api.add_and_include_file_to_latest_version(
-            bundle_name=case_id, file=log_file_path, tags=["gisaid-log", case_id]
+            bundle_name=case_id, file=log_file, tags=[GisaidTag.LOG, case_id]
         )
 
-    def create_gisaid_files_in_housekeeper(self, case_id: str) -> None:
-        """Create all gisaid files in Housekeeper, if needed."""
-        gisaid_samples = self.get_gisaid_samples(case_id=case_id)
-        self.create_gisaid_csv(gisaid_samples=gisaid_samples, case_id=case_id)
-        self.create_gisaid_fasta(gisaid_samples=gisaid_samples, case_id=case_id)
-        self.create_gisaid_log_file(case_id=case_id)
+    def parse_and_get_sars_cov_complementary_reports(
+        self, complementary_report_content_raw: list[dict]
+    ) -> list[GisaidComplementaryReport]:
+        complementary_report_content_raw_unique: list[dict] = remove_duplicate_dicts(
+            complementary_report_content_raw
+        )
+        complementary_report_sars_cov: list[GisaidComplementaryReport] = (
+            self.validate_gisaid_complementary_reports(complementary_report_content_raw_unique)
+        )
+        return self.get_sars_cov_complementary_reports(complementary_report_sars_cov)
+
+    def get_complementary_report_content_raw(self, case_id: str) -> list[dict]:
+        complementary_report: File = self.get_complementary_file_from_hk(case_id)
+        return self.get_complementary_report_content(Path(complementary_report.full_path))
+
+    def create_and_include_gisaid_files_in_hk(self, case_id: str) -> None:
+        """Create and include all GISAID files in Housekeeper."""
+        complementary_report_content_raw: list[dict] = self.get_complementary_report_content_raw(
+            case_id
+        )
+        sars_cov_complementary_reports: list[GisaidComplementaryReport] = (
+            self.parse_and_get_sars_cov_complementary_reports(complementary_report_content_raw)
+        )
```

Comment:
This same logic is done in the `upload_to_gisaid` function. Wouldn't it be better if this function got as a parameter the `sars_cov_complementary_reports` instead of calculating it again?